### PR TITLE
test that LF is allowed in the title attribute and the textarea element

### PR DESCRIPTION
**Pull request by siusin**
These tests validate that:

> If the title attribute’s value contains LF characters, the content is split into multiple lines.

-- LF is allowed, properly rendered in Blink, Webkit and Gecko;

> LF, CR in the API value attribute of the textarea element should be converted to LF and break the line.

-- LF CR is allowed, properly rendered in Blink, Webkit and Gecko;

> in the placeholder attribute value of the textarea element, CR LF character pairs (CRLF) in the hint, as well as all other CR and LF characters in the hint, must be treated as line breaks;

-- LF CR is allowed, properly rendered in Blink,  no line breaks in Webkit and Gecko;

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5272/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/216802138)

# Firefox (nightly channel) #
Testing web-platform-tests at revision e3c36f95f2951fd3bfe78ac9bb37f1da2e6c492d
Using browser at version BuildID 20170330100130; SourceStamp 60d7a0496a3673450ddbc37ec387525148c32604
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5272/html/dom/elements/global-attributes/the-title-attribute-lf.html">/html/dom/elements/global-attributes/the-title-attribute-lf.html</a></summary>


|                                                 Subtest                                                 | Results | Messages |
|---------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                         | OK      |          |
| `the title attribute allows "LF" characters, please manually check if the tooltip is properly rendered` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5272/html/semantics/forms/the-textarea-element/textarea-lf.html">/html/semantics/forms/the-textarea-element/textarea-lf.html</a></summary>


|                                                         Subtest                                                          | Results | Messages |
|--------------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                          | OK      |          |
| `the placeholder attribute of textarea allows "CR" "LF" characters, please manually check if they are properly rendered` | PASS    |          |
| `the value attribute of textarea allows "CR" "LF" characters, please manually check if they are properly rendered`       | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/216802139)

# Chrome (unstable channel) #
Testing web-platform-tests at revision e3c36f95f2951fd3bfe78ac9bb37f1da2e6c492d
Using browser at version 59.0.3053.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5272/html/dom/elements/global-attributes/the-title-attribute-lf.html">/html/dom/elements/global-attributes/the-title-attribute-lf.html</a></summary>


|                                                 Subtest                                                 | Results | Messages |
|---------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                         | OK      |          |
| `the title attribute allows "LF" characters, please manually check if the tooltip is properly rendered` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5272/html/semantics/forms/the-textarea-element/textarea-lf.html">/html/semantics/forms/the-textarea-element/textarea-lf.html</a></summary>


|                                                         Subtest                                                          | Results | Messages |
|--------------------------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                                          | OK      |          |
| `the placeholder attribute of textarea allows "CR" "LF" characters, please manually check if they are properly rendered` | PASS    |          |
| `the value attribute of textarea allows "CR" "LF" characters, please manually check if they are properly rendered`       | PASS    |          |

</details>
</details>

**Comment by ayg** [comment]
Are the title/placeholder portions not adequately tested already by reflection-*.html?

http://w3c-test.org/html/dom/reflection-forms.html

I'm specifically thinking of these tests:

textarea.title: setAttribute() to " \0\x01\x02\x03\x04\x05\x06\x07 \b\t\n\v\f\r\x0e\x0f \x10\x11\x12\x13\x14\x15\x16\x17 \x18\x19\x1a\x1b\x1c\x1d\x1e\x1f foo "
textarea.placeholder: setAttribute() to " \0\x01\x02\x03\x04\x05\x06\x07 \b\t\n\v\f\r\x0e\x0f \x10\x11\x12\x13\x14\x15\x16\x17 \x18\x19\x1a\x1b\x1c\x1d\x1e\x1f foo "

I'm not sure if we have a test that line breaks work in .value.  If you're just testing round-tripping, it would be good to test a wider array of values, but that doesn't have to block acceptance of the test.

File: html/dom/elements/global-attributes/the-title-attribute-lf.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>the title attribute - allows "LF"</title>
+<link rel="help" href="http://w3c.github.io/html/dom.html#the-title-attribute">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+  <h1>the title attribute allows "LF" characters</h1>
+
```
Comment (ayg) [inline html/dom/elements/global-attributes/the-title-attribute-lf.html]: You need to remove the trailing whitespace (here and two other places), or the auto-checking thing won't allow this to be merged.

File: html/dom/elements/global-attributes/the-title-attribute-lf.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>the title attribute - allows "LF"</title>
+<link rel="help" href="http://w3c.github.io/html/dom.html#the-title-attribute">
```
Comment (domenic) [inline html/dom/elements/global-attributes/the-title-attribute-lf.html]: This is not the spec that web platform tests tests.

**Review by domenic — CHANGES_REQUESTED** [review]
Going through old review requests...

In addition to my minor issue, @ayg's review stands with regard to the bigger problems.

**Comment by zcorpan** [comment]
`textarea.value` is already tested by /html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html so I believe what this PR tries to test is already covered by existing tests. So closing. Please let me know if I missed something. :-)

I created a PR to test newlines in `input.value` at https://github.com/w3c/web-platform-tests/pull/6149

If there are rendering differences in the tooltip or so, it's probably better to file bugs for each browser about that. Or possibly creating a reftest for `placeholder` to check that `placeholder="A&#10;B"` is *different* from `placeholder="AB"`

**Comment by cvrebert** [comment]
@zcorpan I don't see any test of `"\0"` in `/the-textarea-element/`, though my search was manual.

**Pull request closed without merging**